src/gc.rs

```diff
@@ -210,10 +210,12 @@ impl Collector {
     fn enter_sweep_phase(&mut self) {
         *self.state.lock().unwrap() = CollectorState::Sweeping;
 
-        for block in unsafe { BLOCK_METADATA.iter().filter(|x| x.map_or(false, |x| x.gc)) } {
-            let obj = unsafe { Gc::from_raw(block.unwrap().ptr as *const OpaqueU8) };
-            if self.colour(obj) == Colour::White {
-                self.dealloc(obj);
+        unsafe {
```

Comment:
Is everything about this `unsafe` including `iter`/`filter`? If not, can we make the `unsafe` part smaller?

Reply:
How about this? 813067bdda01ed6e7031ddbbfe7eeb118176dede